mantidimaging/core/operations/loader.py

```diff
@@ -4,36 +4,40 @@
 import os
 import pkgutil
 import sys
-from typing import List, Protocol, cast, Union, TYPE_CHECKING
-from importlib.machinery import FileFinder, ModuleSpec
-from importlib.abc import Loader
+from importlib.util import module_from_spec
+from typing import List, TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from PyInstaller.loader.pyimod02_importers import FrozenImporter
     from mantidimaging.core.operations.base_filter import BaseFilter
 
-MODULES_OPERATIONS: dict[str, Union['FrozenImporter', Loader]] = {}
-if not MODULES_OPERATIONS:
-    for finder, module_name, _ in pkgutil.walk_packages([os.path.dirname(__file__)]):
+_OPERATION_MODULES_LIST: List[BaseFilter] = []
```

Comment:
Not as part of this PR, but maybe as tech dept in the future if you agree, I feel like this may be better if we place everything here into a singleton so we can move away from the use of private globals

Reply:
I'd say that module level variable is a simple way of achieving a singleton here (python modules are singletons :-) ).

Reply:
I can live with that XD More just a personal preference on my side